src/utils/networks.ts

```diff
@@ -153,80 +159,47 @@ const callContractMethod = async (context: vscode.ExtensionContext) => {
       "contract"
     )) as CompiledJSONOutput;
 
-    if (compiledOutput == undefined) throw errors.ContractNotSelected;
+    if (compiledOutput === undefined) throw errors.ContractNotSelected;
 
     const abi = getAbi(compiledOutput);
-    if (abi == undefined) throw new Error("Abi is not defined.");
+    if (abi === undefined) throw new Error("Abi is not defined.");
 
     const abiItem = await getFunctionInputs(context);
     if (abiItem === undefined) throw new Error("Function is not defined.");
 
-    const params_ = abiItem.inputs?.map((e: any) => e.value);
-    const params = params_ === undefined ? [] : params_;
-
-    logger.success(`Calling ${compiledOutput.name} : ${abiItem.name} -->`);
-
     const contractAddres = getDeployedInputs(context).address;
     if (contractAddres === undefined)
       throw new Error("Enter deployed address of selected contract.");
 
-    if (abiItem.stateMutability === "view") {
-      selectContract(context);
-
-      const contract = new ethers.Contract(
-        contractAddres,
-        abi,
-        getSelectedProvider(context)
-      );
-
-      const result = await contract[abiItem.name as string](...params);
-      logger.success(
-        `Calling ${compiledOutput.name} : ${abiItem.name} --> Success!`
-      );
-      logger.log(JSON.stringify(result));
-    } else {
-      const contract = await getSignedContract(context, contractAddres);
-
-      let result;
-
-      if (abiItem.stateMutability === "nonpayable") {
-        const gasCondition = (await context.workspaceState.get(
-          "gas"
-        )) as string;
-
-        const gasEstimate = await getGasEstimates(gasCondition, context);
-        const settingsGasLimit = (await getConfiguration().get(
-          "gasLimit"
-        )) as number;
-        if (gasEstimate !== undefined) {
-          const maxFeePerGas = (gasEstimate as EstimateGas).price;
-          result = await contract[abiItem.name as string](...params, {
-            gasPrice: ethers.utils.parseUnits(maxFeePerGas.toString(), "gwei"),
-            gasLimit: settingsGasLimit,
-          });
-        } else {
-          result = await contract[abiItem.name as string](...params);
-        }
-      } else {
-        const found: any = abiItem.inputs?.find(
-          (e: any) => e.type === "uint256"
-        );
-        result = await contract[abiItem.name as string](...params, {
-          value: found.value,
-        });
-      }
+    const params_ = abiItem.inputs?.map((e: any) => e.value);
```

Comment:
I think the params should be read inside the mutable & immutable call functions as now the params will contain the payable value.